pkg/apis/serving/v1alpha2/inferenceservice_types.go

```diff
@@ -60,6 +60,29 @@ type DeploymentSpec struct {
 	// This is the up bound for autoscaler to scale to
 	// +optional
 	MaxReplicas int `json:"maxReplicas,omitempty"`
+	// Activate request/response logging
+	// +optional
+	InferenceLogger *InferenceLogger `json:"inferenceLogger,omitempty"`
+}
+
+type InferenceLoggerType string
+
+const (
+	InferenceLogBoth     InferenceLoggerType = "both"
+	InferenceLogRequest  InferenceLoggerType = "request"
+	InferenceLogresponse InferenceLoggerType = "response"
+)
+
+type InferenceLogger struct {
+	// URL to send request logging CloudEvents
+	// +optional
+	Url *string `json:"url,omitempty"`
```

Comment:
Cool -- very simple. Why is this a pointer?

Reply:
I wanted to make clear its optional but checking for empty string is probably ok.

Reply:
I think this is actually the right pattern. If it's optional, it's a pointer.

Reply:
Actually should url be mandatory if InferenceLogger is specified ?